packages/animations/lib/src/modal.dart

```diff
@@ -0,0 +1,208 @@
+/// modal route that will be displayed, such as the enter and exit
+/// transitions, the duration of the transitions, and modal barrier
+/// properties.
+///
+/// The `useRootNavigator` argument is used to determine whether to push the
+/// modal to the [Navigator] furthest from or nearest to the given `context`.
+/// By default, `useRootNavigator` is `true` and the modal route created by
+/// this method is pushed to the root navigator. If the application has
+/// multiple [Navigator] objects, it may be necessary to call
+/// `Navigator.of(context, rootNavigator: true).pop(result)` to close the
+/// modal rather than just `Navigator.pop(context, result)`.
+///
+/// Returns a [Future] that resolves to the value (if any) that was passed to
+/// [Navigator.pop] when the modal was closed.
+///
+/// See also:
+///
+/// * [ModalConfiguration], which is the configuration object used to define
+/// the modal's characteristics.
+Future<T> showModal<T>({
+  @required BuildContext context,
+  @required ModalConfiguration configuration,
+  bool useRootNavigator = true,
+  WidgetBuilder builder,
+}) {
+  assert(configuration != null);
+  assert(useRootNavigator != null);
+  String barrierLabel = configuration.barrierLabel;
+  // Avoid looking up [MaterialLocalizations.of(context).modalBarrierDismissLabel]
+  // if there is no dismissible barrier.
+  if (configuration.barrierDismissible && configuration.barrierLabel == null) {
+    barrierLabel = MaterialLocalizations.of(context).modalBarrierDismissLabel;
+  }
+  assert(!configuration.barrierDismissible || barrierLabel != null);
+  return Navigator.of(context, rootNavigator: useRootNavigator).push<T>(
+    _ModalRoute<T>(
+      barrierColor: configuration.barrierColor,
+      barrierDismissible: configuration.barrierDismissible,
+      barrierLabel: barrierLabel,
+      transitionBuilder: configuration.transitionBuilder,
+      transitionDuration: configuration.transitionDuration,
+      reverseTransitionDuration: configuration.reverseTransitionDuration,
+      builder: builder,
+    ),
+  );
+}
+
+// A modal route that overlays a widget on the current route.
+class _ModalRoute<T> extends PopupRoute<T> {
+  /// Creates a [_ModalRoute] route with the Material fade transition.
+  ///
+  /// [barrierDismissible] is true by default.
+  _ModalRoute({
+    Color barrierColor,
+    bool barrierDismissible = true,
+    String barrierLabel,
+    _ModalTransitionBuilder transitionBuilder,
+    Duration transitionDuration,
+    Duration reverseTransitionDuration,
+    @required this.builder,
+  })  : assert(barrierDismissible != null),
+        _barrierColor = barrierColor,
+        _barrierDismissible = barrierDismissible,
+        _barrierLabel = barrierLabel,
+        _transitionBuilder = transitionBuilder,
+        _transitionDuration = transitionDuration,
+        _reverseTransitionDuration = reverseTransitionDuration;
+
+  @override
+  Color get barrierColor => _barrierColor;
```

Comment:
I believe you can simplify all of these overridden getters in the following way:

In the constructor just do:

```dart
 _ModalRoute({
  this.barrierColor,
  // ...
});
```

... and instead of having the getter and the private field here just do:

```dart
@override
final Color barrierColor;
```